### PR TITLE
docs: Rec. adding/disabling "GenerateAssemblyFileVersion" in .NET project

### PR DESCRIPTION
**Pull request by BinToss**
## Description
In DotNet SDK-style projects, several properties that determine if the compiler will generate AssemblyInfo are default to `true`.
These lead to the build tools creating their own AssemblyInfo.g.cs which conflicts with GitVersion.

1. I'm unsure which header this should go under. Suggestions are welcome.
2. I didn't have much time to test them, but I can confirm that adding these to my project resolved certain build errors. Those errors were caused by the build tools generating an AssemblyInfo.g.cs that overwrote or conflicted with GitVersion's generated AssemblyInfo.g.cs.
3. I may be difficult to contact during the next week, but I'll try to check here frequently.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/input/docs/usage/msbuild.md [hunk 1]
```diff
@@ -50,6 +50,14 @@ The next thing you need to do is to remove the `Assembly*Version` attributes fro
 your `Properties\AssemblyInfo.cs` files. This puts GitVersion.MsBuild in charge of
 versioning your assemblies.
 
+For SDK-style projects, add the following properties set to `false`:
+```
```
Comment (asbjornu) [inline docs/input/docs/usage/msbuild.md:54]:
```suggestion
For SDK-style projects, add the following properties set to `false`:

```xml
```

File: docs/input/docs/usage/msbuild.md [hunk 2]
```diff
@@ -50,6 +50,14 @@ The next thing you need to do is to remove the `Assembly*Version` attributes fro
 your `Properties\AssemblyInfo.cs` files. This puts GitVersion.MsBuild in charge of
 versioning your assemblies.
 
+For SDK-style projects, add the following properties set to `false`:
```
Comment (asbjornu) [inline docs/input/docs/usage/msbuild.md:53]: Can you please explain why these properties need to be set to `false`?

**Review by asbjornu — CHANGES_REQUESTED** [review]

File: docs/input/docs/usage/msbuild.md (hunk 2 above)
Comment (BinToss) [reply docs/input/docs/usage/msbuild.md:53]: img001
From https://elanderson.net/2017/06/duplicate-system-reflection-assemblycompanyattribute-attribute/

I has actually saw this screenshot I saved on my phone a few minutes and remembered to add it here.

**Comment by BinToss** [comment]
I'm currently moving to a new house, so it will be a few days (at worst) before I can polish this PR.

File: docs/input/docs/usage/msbuild.md (hunk 2 above)
Comment (asbjornu) [reply docs/input/docs/usage/msbuild.md:53]: Thanks, but sorry that I wasn't clear: I was hoping you could describe it in the documentation (i.e. in this PR) and not just to me directly. :)

Comment (BinToss) [reply docs/input/docs/usage/msbuild.md:53]: Found some time to work on this again.
Sadly, there's nothing in the Microsoft Documentation that specifies the need to disable these property controls in favor of an alternative AssemblyInfo generator.
Aside from using an [Archive.org link](https://web.archive.org/web/20200918071836/https://elanderson.net/2017/06/duplicate-system-reflection-assemblycompanyattribute-attribute/), would you recommend any specific way of referring to Eric L. Anderson's post?

**Comment by BinToss** [comment]
Honestly, I'm rethinking the scope of this Pull Request now that I know the GitVersion.Core (GitVersion.MSBuild, but for DotNet SDK Core projects) package exists.

File: docs/input/docs/usage/msbuild.md (hunk 2 above)
Comment (asbjornu) [reply docs/input/docs/usage/msbuild.md:53]: We can just refer to [Microsoft's documentation](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#generateassemblyconfigurationattribute), perhaps?

Comment (BinToss) [reply docs/input/docs/usage/msbuild.md:53]: Sure. Completed in https://github.com/BinToss/GitVersion/commit/8b969a3be6b523ea93f7ff0700eb6305a2a2371a

**Comment by BinToss** [comment]
Dropped in favor of #2841 
reason(s): conforming to branch name scheme

**Pull request closed without merging**